component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -809,15 +761,51 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
             if (httpConnection != null) {
                 httpConnection.disconnect();
             }
-            if (httpsConnection != null) {
-                httpsConnection.disconnect();
-            }
         }
         return false;
     }
 
     /**
-     * Get a screen value from the user attributes.
+     * Proceed with SMS API's rest call.
+     *
+     * @param context      the AuthenticationContext
+     * @param smsUrl       the smsUrl
+     * @param httpMethod   the httpMethod
+     * @param headerString the headerString
+     * @param payload      the payload
+     * @param httpResponse the httpResponse
+     * @param mobile       the mobile number
+     * @param otpToken     the OTP token
+     * @return true or false
+     * @throws IOException
+     * @throws AuthenticationFailedException
+     */
+    public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String httpMethod,
+                                String headerString, String payload, String httpResponse, String mobile,
+                                String otpToken) throws IOException, AuthenticationFailedException {
+        boolean connection;
+        String smsMessage = SMSOTPConstants.SMS_MESSAGE;
+        URLEncoder encoder = new URLEncoder();
+        String encodedMobileNo = encoder.encode(mobile);
+        smsUrl = smsUrl.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg",
+                smsMessage.replaceAll("\\s", "+") + otpToken);
+        URL smsProviderUrl = new URL(smsUrl);
+        HttpURLConnection httpConnection = (HttpURLConnection) smsProviderUrl.openConnection();
```

Comment:
Move this to else block. You are opening the connection two times for https flow.